### PR TITLE
Add base config files to support Web Sockets

### PR DESCRIPTION
**Pull request by peenyaa7**
Contains some examples like this:

image

### CODE REVIEW

File: src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   * This method registers the endpoint for the WebSocket.
+   *
+   * @param registry the registry of the Stomp endpoints.
+   */
+  @Override
+  public void registerStompEndpoints(StompEndpointRegistry registry) {
+    registry
+        .addEndpoint("/preguntonic") // The endpoint for the WebSocket.
+        .setAllowedOriginPatterns("*") // The allowed origins.
+        .withSockJS(); // Enable SockJS -> Add support to clients that don't support WebSocket.
```
Comment (JFCowboy) [inline src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java:29]: 👍  🚀 
Question about this 2 lines:

1.  when we move to "Production" the `setAllowedOriginPatterns` should it be removed?
2. Meanwhile I'm trying to understand better the `.withSockJS()` line I found this [spring-boot-websocket-without-sockjs](https://stackoverflow.com/questions/67920378/spring-boot-websocket-without-sockjs/68076180#68076180) question looks like it is for support `http` in the connection for the clients and without this line you can connect using `ws`. So do you think we need to add support for the other connection (`ws`)?
3. Just for curiosity do you know how to enable secure protocole? (`https`, `wss`)

File: src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            "/room"); // [2]Enable the simple broker (some examples of destinations).
+
+    // Help:
+    // [1] The prefix for the application destination is used to filter the messages
+    // that are sent
+    // from the client to the server.
+    // [2] Enable a simple broker in memory to send messages from the server to the
+    // client. The
+    // destinations are used to filter the messages.
+    // Any message that doesn't match with any of the destinations will be ignored.
```
Comment (JFCowboy) [inline src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java]: Style suggestion, sorry in some cases spotless doesn't work well with line breaks in comments.
```suggestion
    // Help:
    // [1] The prefix for the application destination is used to filter the messages
    // that are sent from the client to the server.
    // [2] Enable a simple broker in memory to send messages from the server to the
    // client. The destinations are used to filter the messages.
    // Any message that doesn't match with any of the destinations will be ignored.
```

File: src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    /**
+     * HELP:
+     *
+     * <p>[1] The annotation @MessageMapping("/message") is used to map the message to the
+     * destination "app/message" (the prefix is defined in the WebSocketConfig). [2] The
+     * annotation @SendTo("/room/room1") is used to send the message to the topic "/room/room1".
+     */
+    System.out.println(
+        "New message was received: " + message + ". Sending it to the topic /room/room1...");
```
Comment (JFCowboy) [inline src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java:35]: WDYT about use [slf4j](https://www.baeldung.com/slf4j-with-log4j2-logback) for logs?
Lombok label: https://projectlombok.org/features/log

File: src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  @SendTo("/room/room1") // [2]
+  public String example(String message) {
+
+    /**
+     * HELP:
+     *
+     * <p>[1] The annotation @MessageMapping("/message") is used to map the message to the
+     * destination "app/message" (the prefix is defined in the WebSocketConfig). [2] The
+     * annotation @SendTo("/room/room1") is used to send the message to the topic "/room/room1".
+     */
```
Comment (JFCowboy) [inline src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java:33]: It is a good comment but is a `Dangling Javadoc comment` should we move this comment to the method comment?

File: src/main/java/com/devathon/preguntonic/services/WebSockerService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Service
+@EnableScheduling
+public class WebSockerService {
```
Comment (JFCowboy) [inline src/main/java/com/devathon/preguntonic/services/WebSockerService.java:16]: Typo
```suggestion
public class WebSocketService {
```

**Review by JFCowboy — APPROVED** [review]
👍  Approve with comments.

Most of the comments are style suggestions, but WDYT about use a logger instead of STDOUT

**Review by Gaston-Herrlein — APPROVED** [review]

File: src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java (hunk 1 above)
Comment (peenyaa7) [reply src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java:29]: 1. Should be replaced by our own domain (example.com). Althought, if we want to open connection to internet, we can keep this line
2. Yes! We can add web socket support and keep http support, so i will add web socket support.
3. We cant add secure protocols from backend code, we need to deploy on a remote server and install lets encrypt certificates

File: src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java (hunk 3 above)
Comment (peenyaa7) [reply src/main/java/com/devathon/preguntonic/controllers/WebSocketController.java:35]: yes, we should use @Slf4j, i always use that, so we have to add lombok support

File: src/main/java/com/devathon/preguntonic/services/WebSockerService.java (hunk 5 above)
Comment (peenyaa7) [reply src/main/java/com/devathon/preguntonic/services/WebSockerService.java:16]: I have to change that using vscode (its needed to change file name)

**Pull request merged**